src/Shared/FileUtilities.cs

```diff
@@ -930,5 +946,21 @@ private static string CurrentExecutableOverride
                 return s_currentExecutableOverride;
             }
         }
+
+        /// <summary>
+        /// Returns true when the entry point application is Visual Studio.
+        /// </summary>
+        internal static bool RunningInVisualStudio
+        {
+            get
+            {
+                // Check if initialized and do so if not yet
+                if (!s_runningInVisualStudio.HasValue)
+                {
+                    GetTestExecutionInfo();
```

Comment:
Does this need to worry about thread safety or maybe preventing multiple of these running in parallel, from multiple projects being built in parallel? Maybe use `Lazy<T>` if so?